share/eds/store.go

```diff
@@ -298,8 +297,11 @@ func (s *Store) getAccessor(ctx context.Context, key shard.Key) (*dagstore.Shard
 }
 
 func (s *Store) getCachedAccessor(ctx context.Context, key shard.Key) (*accessorWithBlockstore, error) {
-	// try to fetch from cache
-	accessor, err := s.cache.Get(key)
+	lk := &s.cache.stripedLocks[shardKeyToStriped(key)]
+	lk.Lock()
+	defer lk.Unlock()
```

Comment:
Is there a way to prevent the Store from accessing the cache's internal fields?

Reply:
Not that I could think of, I thought about it for quite a long time. Was hoping you would have a better idea that would still be clean

Reply:
Let's park that caching EDSes on write/ new mount type as a new issue?

Reply:
SGTM